src/ripple/basics/safe_cast.h

```diff
@@ -73,6 +73,56 @@ safe_cast(Src s) noexcept
     return safe_cast<Dest>(static_cast<std::underlying_type_t<Src>>(s));
 }
 
+// unsafe_cast explicitly flags a static_cast as not necessarily able to hold
+// all values of the source. It includes a compile-time check so that if
+// underlying types become safe, it can be converted to a safe_cast.
+
+template <class Dest, class Src>
+inline
+constexpr
+std::enable_if_t
+<
+    std::is_integral<Dest>::value && std::is_integral<Src>::value,
+    Dest
+>
+unsafe_cast(Src s) noexcept
```

Comment:
It can also go from larger type to smaller. I named it `unsafe_cast` as a contrast to `safe_cast`, and specifically to draw attention to it: "This is not safe. Make sure you're doing the right thing." If you have a suggestion for a better name, I'm game.